src/utils/formatSearchResponse.ts

```diff
@@ -54,15 +54,86 @@ const formatSearchResponse = async (
   return finalResponse;
 };
 
-const getOtherDetails = async (searchItem: Record<string, any>): Promise<Record<string, string>> => {
+const getLicenseConstraints = (searchItem: Record<string, any>): string => {
+  let licenseConstraints = '';
+
+  if (searchItem?._source?.MD_LegalConstraintsOtherConstraintsObject?.[0]?.default) {
+    licenseConstraints = `${searchItem?._source?.MD_LegalConstraintsOtherConstraintsObject?.[0]?.default} <br>`;
+  }
+  if (searchItem?._source?.MD_LegalConstraintsOtherConstraintsObject?.[2]?.text) {
+    licenseConstraints =
+      licenseConstraints + `${searchItem?._source?.MD_LegalConstraintsOtherConstraintsObject?.[2]?.text}`;
+  }
+
+  return licenseConstraints;
+};
+
+const getLimitationPublicAccess = (searchItem: Record<string, any>): string => {
+  let limitationPublicAccess = '';
+
+  if (searchItem?._source?.cl_accessConstraints?.[0]?.default) {
+    limitationPublicAccess = `${searchItem?._source?.cl_accessConstraints?.[0]?.default} <br>`;
+  }
+  if (searchItem?._source?.cl_accessConstraints?.[0]?.text) {
+    limitationPublicAccess = limitationPublicAccess + `${searchItem?._source?.cl_accessConstraints?.[0]?.text} <br>`;
+  }
+  if (searchItem?._source?.licenseObject?.[0]?.defasult) {
+    limitationPublicAccess = limitationPublicAccess + `${searchItem?._source?.licenseObject?.[0]?.default}`;
+  }
+
+  return limitationPublicAccess;
+};
+
+const getPublishedBy = (publishedBy: Record<string, any>): string => {
+  let dataOwner = '';
+
+  if (publishedBy.role) {
+    dataOwner = `${publishedBy.role}, `;
+  }
+  if (publishedBy.organisationValue) {
+    dataOwner += `${publishedBy.organisationValue} <br>`;
+  }
+  if (publishedBy.email) {
+    dataOwner += `${publishedBy.email}`;
+  }
+
+  return dataOwner;
+};
+
+const getCatalogueDate = (searchItem: Record<string, any>): string => {
   const catalogueDate = searchItem?._source?.dateStamp ? new Date(searchItem?._source?.dateStamp) : '';
+  return catalogueDate
```

Comment:
Can we not use the formatDate function that already exists and make the necessary changes without affecting the existing functionality?

Reply:
Addressed.